src/sentry/tasks/data_export.py

```diff
@@ -1,8 +1,52 @@
 from __future__ import absolute_import
+
+import csv
+from datetime import datetime
+
+from sentry.api import client
+from sentry.constants import ExportQueryType
 from sentry.tasks.base import instrumented_task
 
 
 @instrumented_task(name="sentry.tasks.data_export.compile_data", queue="data_export")
 def compile_data(data_export):
-    # TODO(Leander): Implement logic to compile CSV data via provided ExportedData object
+    if data_export.query_type == ExportQueryType.DISCOVER_V1:
+        process_discover_v1(data_export)
+
+    elif data_export.query_type == ExportQueryType.BILLING_REPORT:
+        # TODO(Leander): Implement logic for billing report CSVs
+        return
+    elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
+        # TODO(Leander): Implement logic for issue tag CSVs
+        return
```

Comment:
This is the best way I could come up with to abstract this 'Export to CSV' task.

 Each use case will have different processing logic and by massaging the payloads, we can use the same `convert_to_csv` function (and eventually, `upload_to_filestore` function) for all of them.